Neos.Flow/Classes/ObjectManagement/DependencyInjection/PropertyInjectionTrait.php

```diff
@@ -27,14 +27,18 @@ trait PropertyInjectionTrait
      * @param callable $lazyInjectionResolver
      * @return void
      */
-    private function Flow_Proxy_LazyPropertyInjection($propertyObjectName, $propertyClassName, $propertyName, $setterArgumentHash, callable $lazyInjectionResolver)
+    private function Flow_Proxy_LazyPropertyInjection(string $propertyObjectName, string $propertyClassName, string $propertyName, string $setterArgumentHash, callable $lazyInjectionResolver)
     {
         $injection_reference = &$this->$propertyName;
         $this->$propertyName = \Neos\Flow\Core\Bootstrap::$staticObjectManager->getInstance($propertyObjectName);
         if ($this->$propertyName === null) {
             $this->$propertyName = \Neos\Flow\Core\Bootstrap::$staticObjectManager->getLazyDependencyByHash($setterArgumentHash, $injection_reference);
             if ($this->$propertyName === null) {
                 $this->$propertyName = \Neos\Flow\Core\Bootstrap::$staticObjectManager->createLazyDependency($setterArgumentHash, $injection_reference, $propertyClassName, $lazyInjectionResolver);
```

Comment:
i finally kindof find the reson for this buggy behavior that on the fly di changes dont work correctly:
https://github.com/neos/flow-development-collection/pull/2701#pullrequestreview-921001408

it seems that this method `createLazyDependency` is returning `null` and the next call: $staticObjectManager->get($propertyObjectName);

doesnt return an object of the proxied class but rather an object based of the class in my working dir...

Reply:
so https://github.com/neos/flow-development-collection/pull/2701#pullrequestreview-921001408 even shorter: 
 go to a lazy dependency injection (in the old doc comment style - so you can compare this exact behaviour with `flow@master`)

- remove the `@Flow\Inject` and run a flow
- add the  `@Flow\Inject` again and run flow again -> now the property should be injected.

but at `flow@robert` as i seems this code:

https://github.com/neos/flow-development-collection/blob/aea746af48b02828cec7017aabeb278d5bf5a1f6/Neos.Flow/Classes/ObjectManagement/ObjectManager.php#L420

is triggered but `flow@master` is working in this scenario